File: MsaCookingApp.Contracts/Features/Meals/DTOs/SpoonacularGetMealIngredient.cs
namespace MsaCookingApp.Contracts.Features.Meals.DTOs;

public class SpoonacularGetMealIngredient
{
    public required string Id { get; set; }
    public required string Name { get; set; }
    public string? Original { get; set; }
    public double? Amount { get; set; }
    public string? Unit { get; set; }
}

File: msa_cooking_app_client/lib/features/meals/models/spoonacular_get_meal_ingredient.dart
import 'package:json_annotation/json_annotation.dart';

part 'spoonacular_get_meal_ingredient.g.dart';

@JsonSerializable()
class SpoonacularGetMealIngredient {
  final String id;
  final String name;
  final String? original;
  final double? amount;
  final String? unit;

  SpoonacularGetMealIngredient(this.id, this.name, this.original, this.amount, this.unit);

  factory SpoonacularGetMealIngredient.fromJson(Map<String, dynamic> json) =>
      _$SpoonacularGetMealIngredientFromJson(json);

  Map<String, dynamic> toJson() => _$SpoonacularGetMealIngredientToJson(this);
}